qiskit_ode/signals/signals.py

```diff
@@ -247,8 +279,53 @@ def __init__(
 
         self._start_time = start_time
 
-        self._carrier_freq = Array(carrier_freq)
-        self._phase = Array(phase)
+        # initialize internally stored carrier/phase information
+        self._carrier_freq = None
+        self._phase = None
+        self._carrier_arg = None
+        self._phase_arg = None
+
+        # set carrier and phase
+        self.carrier_freq = carrier_freq
+        self.phase = phase
+
+    @classmethod
+    def from_Signal(
+        cls,
+        signal: Signal,
+        dt: float,
+        n_samples: int,
+        start_time: Optional[float] = 0.0,
+        sample_carrier: Optional[bool] = False,
+    ):
+        """
+        Constructs a ``DiscreteSignal`` signal object by sampling a ``Signal``.
+
+        Args:
+            signal: Signal to sample.
+            dt: Time increment to use.
+            n_samples: Number of steps to resample with.
+            start_time: Start time from which to resample.
+            sample_carrier: Whether or not to keep the carrier analog or include it in the
+                            sampling.
```

Comment:
`keep the carrier along` is a bit vague

Reply:
Updated this, as well as the corresponding text in `DiscreteSignalSum.from_SignalSum` to be more explicit.